### PR TITLE
Upgrade SDK, Firebase, null_safety

### PR DESCRIPTION
**Pull request by torstenek**
Upgraded dependencies to SDK >=2.12.0 and Firebase_core ^1.0.0 along with required transitive dependencies. Works with SDK 2.13.0-77.0.dev

### CODE REVIEW

File: lib/src/firebase_image.dart [hunk 1]
```diff
@@ -35,18 +35,19 @@ class FirebaseImage extends ImageProvider<FirebaseImage> {
...
-  /// [firebaseApp] Default: the default Firebase app. Specifies a custom Firebase app to make the request to the bucket from (optional)
+  /// [firebaseApp] Specifies a custom Firebase app to make the request to the bucket from (required)
   FirebaseImage(
     String location, {
     this.shouldCache = true,
     this.scale = 1.0,
     this.maxSizeBytes = 2500 * 1000, // 2.5MB
     this.cacheRefreshStrategy = CacheRefreshStrategy.BY_METADATA_DATE,
-    this.firebaseApp,
+    required this.firebaseApp,
```
Comment (awhitford) [inline lib/src/firebase_image.dart]: This breaking change should be documented in the change log and/or readme.

**Comment by awhitford** [comment]
The top-level `pubspec.lock` file should be _removed_ (and added to `.gitignore`).  It should not be committed for a library, only for an app.

See https://github.com/toptal/gitignore/blob/2d8160c7f607e142929d2ba213620a638e50603f/templates/Flutter.gitignore#L16

**Comment by awhitford** [comment]
Actually, I think PR #30 is slightly better because it leverages the `late` keyword.

File: lib/src/firebase_image.dart (hunk 1 above)
Comment (awhitford) [reply lib/src/firebase_image.dart]: Is this _breaking change_ necessary?

Comment (torstenek) [reply lib/src/firebase_image.dart]: Agreed. firebaseApp is no longer required.

**Comment by torstenek** [comment]
Using the late keyword is better. Changed it. Sorry for not spotting this in an earlier PR.

**Comment by torstenek** [comment]
I'll leave this PR here since this is the version I'm using, and in the event that it can be useful. I suppose it would be obviated by a merge of PR #30. As with PR #30 it would probably benefit from further testing.

**Comment by mattreid1** [comment]
Just merged PR #30. Thanks for your help though!

**Comment by torstenek** [comment]
NP 👍

**Pull request closed without merging**